streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java

```diff
@@ -80,7 +81,7 @@ public void setUp() {
         final RocksDBSegmentedBytesStore root =
             new RocksDBSegmentedBytesStore("test", "metrics-scope", 0L, SEGMENT_INTERVAL, schema);
         final RocksDBSessionStore sessionStore = new RocksDBSessionStore(root);
-        cachingStore = new CachingSessionStore(sessionStore, SEGMENT_INTERVAL);
+        cachingStore = new CachingSessionStore(sessionStore, SEGMENT_INTERVAL, true);
```

Comment:
If you set the flag to `true` here, you only test the DSL mode of the store, right? The processor API mode is not tested. I think you should test both modes. You could verify the expected timestamps for each mode in two separate test methods with each method instantiating the respective store. For all other expected results you could use the same test methods and run them twice, once with a store in DSL mode and once with the store in Processor API mode.